File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Config/Encoding/StringEncodings.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Config.Encoding;
using Db4objects.Db4o.Internal.Encoding;

namespace Db4objects.Db4o.Config.Encoding
{
	/// <summary>All built in String encodings</summary>
	/// <seealso cref="Db4objects.Db4o.Config.IConfiguration.StringEncoding(IStringEncoding)
	/// 	">Db4objects.Db4o.Config.IConfiguration.StringEncoding(IStringEncoding)</seealso>
	public class StringEncodings
	{
		public static IStringEncoding Utf8()
		{
			return new UTF8StringEncoding();
		}

		public static IStringEncoding Unicode()
		{
			return new UnicodeStringEncoding();
		}

		public static IStringEncoding Latin()
		{
			return new LatinStringEncoding();
		}
	}
}

File: db4oj/core/src/com/db4o/config/encoding/StringEncodings.java
/* Copyright (C) 2008  Versant Inc.  http://www.db4o.com */

package com.db4o.config.encoding;

import com.db4o.config.*;
import com.db4o.internal.encoding.*;

/**
 * All built in String encodings
 * @see Configuration#stringEncoding(StringEncoding)
 */
public class StringEncodings {
	
	public static StringEncoding utf8() {
		return new UTF8StringEncoding();
	}
	
	public static StringEncoding unicode() {
		return new UnicodeStringEncoding();
	}
	
	public static StringEncoding latin() {
		return new LatinStringEncoding();
	}

}
